### ISSUE TITLE
Check for empty password is very awkward to circumvent

### ISSUE THREAD

**Issue by DavidAnderson684** [woocommerce/woocommerce#13689]
## EXPLANATION OF THE ISSUE

I am currently implementing a solution for password-less login, using a different, more convenient, method.

WC_Form_Handler::process_login() includes a check `if ( empty( $_POST['password'] ) ) {` which is rather awkward/inelegant to circumvent. WooCommerce itself doesn't do any password checking - it out-sources this to other parts of WordPress which are reasonably easy to hook. So, the combinations of its insistence on a non-empty password here and the difficulty in over-riding (there's nothing obvious in the context), makes life awkward.

Suggestion solution: add an additional check `&& apply_filters('woocommerce_process_login_require_password', true)` check on that line.

**Comment by DavidAnderson684** [comment]
For more flexibility also, the mentioned check should be moved **after** the immediately following fragment that gets a username from the email address, and the username passed as a parameter to the suggested filter. That way, the developer who wants to make the password optional can make it a per-user thing without duplicating this username-setting fragment.

**Comment by mikejolley** [comment]
Our login form has a password field, so yes we check that here too. 

Look at form-login.php. This cannot be removed with filters.

Sounds like you need your own form + handler.

**Comment by mikejolley** [comment]
Since this handler is for a specific form with a password field, I don't think we need to do anything here. If you have extra info about this, or more justification, please reply or submit a PR proposal.

**Comment by DavidAnderson684** [comment]
Hi Mike,

I've created a pull request now. The use case is that I want to scan a QR code on a smartphone app as an alternative to the password. The password field remains there for normal use (i.e. it's either/or, not replacement).

Here's a copy-paste of the blurb on the pull request. In it, I've changed the suggestion - since WP already performs its own empty-password check, the redundant/repeated/awkward check here should just be removed. WP's check is easy for developers to deal with - the one in WC here is not.

...

WooCommerce here checks that the password is not empty. This check is redundant, because it is already performed by wp_authenticate_email_password() (which is invoked, eventually, by the call in WC to wp_signon() ). i.e. When you cut these lines, in the absence of other hooks, you will still get an error from WordPress about the missing password. Moreover, the check is harmful, because it prevents extension of the WC login form. For example, in my use case, I have added a code to be scanned by a smartphone app as an alternative to the password. When the code is scanned, a password is required. I could build my own replacement login form that duplicates large chunks of WC, but all I'd be doing is removing this single check. (I want to keep the rest of the form so that I remain compatible with other things that hook in... for example, I can still use a separate OTP-based two-factor plugin that adds a requirement for a TOTP code - that is tested and working).

This extra/redundant check for an empty password on the WooCommerce form prevents my already-working-on-the-standard-wp-login-form code from working here, and there's no way to work around it that isn't hacky/ugly (e.g. direct manipulation of $_POST).
